src/utils/diff.ts

```diff
@@ -0,0 +1,54 @@
+import { set, isEmpty, isEqual, isNil as isNullOrUndefined } from 'lodash';
+import { Dictionary } from '../configuration/interfaces';
+
+const OBJECTS_EMPTY_ERROR                = 'objects should be not empty';
+const OBJECTS_DIFFERENT_PROPERTIES_ERROR = 'objects should contain the same properties';
+
+function getFullPropertyPath (property: string, parentProperty: string): string {
+    if (parentProperty)
+        return `${parentProperty}.${property}`;
+
+    return property;
+}
+
+function validate (source: Dictionary<object>, modified: Dictionary<object>): void {
+    if (isEmpty(source) || isEmpty(modified))
+        throw new TypeError(OBJECTS_EMPTY_ERROR);
```

Comment:
We use the `GeneralError` class for argument validation errors.
* https://github.com/DevExpress/testcafe/blob/master/src/index.js#L19
* https://github.com/DevExpress/testcafe/blob/master/src/index.js#L32
and so on.